modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -99,11 +126,24 @@ const emptyMultiTableCells = (
   endTableCells: SugarElement<HTMLTableCellElement>[],
   betweenRng: Range
 ): boolean => {
-  cleanCells(startTableCells.concat(endTableCells));
+  const rng = editor.selection.getRng();
+
+  const startCell = startTableCells[0];
+  const endCell = endTableCells[endTableCells.length - 1];
+
+  deleteContentInsideCell(startCell, rng, true);
+  deleteContentInsideCell(endCell, rng, false);
+
+  // Only clean empty cells, the first and last cells have the potential to still have content
+  const startTableCellsToClean = editor.dom.isEmpty(startCell.dom) ? startTableCells : startTableCells.slice(1);
+  const endTableCellsToClean = editor.dom.isEmpty(endCell.dom) ? endTableCells : endTableCells.slice(0, -1);
```

Comment:
Should this be using `Empty.isEmpty` instead? That's what we use above in `emptySingleTableCells`.
```suggestion
  const startTableCellsToClean = Empty.isEmpty(startCell.dom) ? startTableCells : startTableCells.slice(1);
  const endTableCellsToClean = Empty.isEmpty(endCell) ? endTableCells : endTableCells.slice(0, -1);
```